features/FEATURE_COMMON_PAL/mbed-trace/source/mbed_trace.c

```diff
@@ -92,6 +92,14 @@
 #define DEFAULT_TRACE_CONFIG              TRACE_MODE_COLOR | TRACE_ACTIVE_LEVEL_ALL | TRACE_CARRIAGE_RETURN
 #endif
 
+/** static trace configuration **/
+#ifdef MBED_TRACE_STATIC
+static uint8_t trace_filters_exclude[DEFAULT_TRACE_FILTER_LENGTH];
+static uint8_t trace_filters_include[DEFAULT_TRACE_FILTER_LENGTH];
+static uint8_t trace_line[DEFAULT_TRACE_LINE_LENGTH];
+static uint8_t trace_tmp_data[DEFAULT_TRACE_TMP_LINE_LEN];
+#endif
+
 /** default print function, just redirect str to printf */
 static void mbed_trace_realloc( char **buffer, int *length_ptr, int new_length);
```

Comment:
It could be useful to guard mbed_trace_malloc and mbed_trace_realloc functions to ensure they're not used when MBED_TRACE_STATIC is set?